sklearn/tests/test_calibration.py

```diff
@@ -618,3 +621,176 @@ def test_calibration_votingclassifier():
     calib_clf = CalibratedClassifierCV(base_estimator=vote, cv="prefit")
     # smoke test: should not raise an error
     calib_clf.fit(X, y)
+
+
+@pytest.fixture(scope="module")
+def iris_data():
+    return load_iris(return_X_y=True)
+
+
+@pytest.fixture(scope="module")
+def iris_data_binary(iris_data):
+    X, y = iris_data
+    return X[y < 2], y[y < 2]
+
+
+def test_calibration_display_reg(pyplot, iris_data):
+    X, y = iris_data
+    reg = LinearRegression().fit(X, y)
+
+    msg = "'estimator' should be a fitted classifier"
+    with pytest.raises(ValueError, match=msg):
+        CalibrationDisplay.from_estimator(reg, X, y)
+
+
+def test_calibration_display_no_predict_proba(pyplot, iris_data_binary):
+    X, y = iris_data_binary
+    clf = LinearSVC().fit(X, y)
+
+    msg = "response method predict_proba is not defined in"
+    with pytest.raises(ValueError, match=msg):
+        CalibrationDisplay.from_estimator(clf, X, y)
+
+
+def test_calibration_display_not_fitted(pyplot, iris_data_binary):
+    X, y = iris_data_binary
+    clf = LogisticRegression()
+
+    with pytest.raises(NotFittedError):
+        CalibrationDisplay.from_estimator(clf, X, y)
```

Comment:
I would merge these checks in the same function called `test_calibration_display_validation` similarly to https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/metrics/_plot/tests/test_confusion_matrix_display.py#L28